From 0acf28e85ffa8aed05fa07b759ef8453ff8652ec Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Wed, 16 Mar 2016 14:50:59 +0200
Subject: [PATCH 0961/1240] fix: clock: ap806: fix the clock of Tclock of
 AP-806 SoC to be 200MHz

- For the new revision of the AP-806, use new clock fixed to 200MHz

Change-Id: I85c12ddab178b03e5d31d47ebfa271bd226a362d
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28291
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Haim Boot <hayim@marvell.com>
---
 arch/arm/cpu/armv8/armada8k/clock.c | 6 ++++--
 include/configs/armada8k.h          | 1 +
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/arch/arm/cpu/armv8/armada8k/clock.c b/arch/arm/cpu/armv8/armada8k/clock.c
index 38f7e4d..d7a563d 100644
--- a/arch/arm/cpu/armv8/armada8k/clock.c
+++ b/arch/arm/cpu/armv8/armada8k/clock.c
@@ -39,10 +39,12 @@ u32 soc_mss_clk_get(void)
 
 u32 soc_tclk_get(void)
 {
-#ifndef CONFIG_PALLADIUM
+#ifdef CONFIG_PALLADIUM
+	return CONFIG_MSS_FREQUENCY;
+#elif defined CONFIG_AP806_Z_SUPPORT
 	return soc_mss_clk_get();
 #else
-	return CONFIG_MSS_FREQUENCY;
+	return CONFIG_TCLK_FREQUENCY;
 #endif
 }
 
diff --git a/include/configs/armada8k.h b/include/configs/armada8k.h
index 3644cc4..6127e46 100644
--- a/include/configs/armada8k.h
+++ b/include/configs/armada8k.h
@@ -46,6 +46,7 @@
 #define CONFIG_REMAKE_ELF
 
 /* Clock Definitions */
+#define CONFIG_TCLK_FREQUENCY	200 * MHz
 #ifndef CONFIG_PALLADIUM
 #define COUNTER_FREQUENCY	(25 * 1000000)
 #else
-- 
1.9.1

